File: java_integration/app/src/main/java/com/firewok/example/java_integration/feedresource/sku/SkuActivity.java
package com.firewok.example.java_integration.feedresource.sku;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MenuItem;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import com.firework.common.feed.FeedResource;
import com.firewok.example.java_integration.BuildConfig;
import com.firewok.example.java_integration.R;
import com.firewok.example.java_integration.databinding.ActivitySkuBinding;
import com.firework.videofeed.FwVideoFeedView;
import com.firework.viewoptions.BaseOption;
import com.firework.viewoptions.ViewOptions;
import java.util.Arrays;
import java.util.List;

public class SkuActivity extends AppCompatActivity {
    private ActivitySkuBinding binding;
    private static final List<String> DEFAULT_IDS = Arrays.asList("testSku1", "testSku2");

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivitySkuBinding.inflate(LayoutInflater.from(this));
        setContentView(binding.getRoot());
        
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(getString(R.string.sku_screen_title));
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setDisplayShowHomeEnabled(true);
        }

        setupDetails();
        initVideoFeedView();

        binding.etSkuIds.setText(String.join(",", DEFAULT_IDS));
        binding.btnApplyFilter.setOnClickListener(v -> {
            String[] productIds = binding.etSkuIds.getText().toString().split(",");
            List<String> productIdList = Arrays.asList(productIds);
            updateViewOptions(productIdList);
        });
    }

    @SuppressLint("SetTextI18n")
    private void setupDetails() {
        binding.details.source.setText("Sku");
        binding.details.channel.setText(BuildConfig.FW_CHANNEL_ID);
        binding.details.playlist.setText("N/A");
    }

    private void initVideoFeedView() {
        FwVideoFeedView videoFeedView = binding.fwVideoFeedView;

        ViewOptions viewOptions = new ViewOptions.Builder()
            .baseOption(new BaseOption.Builder()
                .feedResource(new FeedResource.Sku(BuildConfig.FW_CHANNEL_ID, DEFAULT_IDS))
                .build())
            .build();

        videoFeedView.init(viewOptions);
    }

    private void updateViewOptions(List<String> productIds) {
        binding.fwVideoFeedView.destroy();
        binding.feedContainer.removeAllViews();
        
        FwVideoFeedView videoFeedView = new FwVideoFeedView(this);
        ViewOptions viewOptions = new ViewOptions.Builder()
            .baseOption(new BaseOption.Builder()
                .feedResource(new FeedResource.Sku(BuildConfig.FW_CHANNEL_ID, productIds))
                .build())
            .build();
        
        binding.feedContainer.addView(videoFeedView);
        videoFeedView.init(viewOptions);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    public static Intent getIntent(Context context) {
        return new Intent(context, SkuActivity.class);
    }
} 

File: feed_resources/app/src/main/java/com/firework/example/feedresources/sku/SkuActivity.kt
package com.firework.example.feedresources.sku

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import androidx.appcompat.app.AppCompatActivity
import com.firework.common.feed.FeedResource
import com.firework.example.feedresources.BuildConfig.FW_CHANNEL_ID
import com.firework.example.feedresources.R
import com.firework.example.feedresources.databinding.ActivitySkuBinding
import com.firework.videofeed.FwVideoFeedView
import com.firework.viewoptions.baseOptions
import com.firework.viewoptions.viewOptions

class SkuActivity : AppCompatActivity() {
    private lateinit var binding: ActivitySkuBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySkuBinding.inflate(LayoutInflater.from(this))
        setContentView(binding.root)
        setTitle(R.string.sku_screen_title)

        setupDetails()

        initVideoFeedView()

        binding.etSkuIds.setText(DEFAULT_IDS.joinToString(","))
        binding.btnApplyFilter.setOnClickListener {
            val productId = binding.etSkuIds.text.toString().split(",")
            updateViewOptions(productId)
        }
    }

    @SuppressLint("SetTextI18n")
    private fun setupDetails() {
        binding.details.source.text = "Sku"
        binding.details.channel.text = FW_CHANNEL_ID
        binding.details.playlist.text = "N/A"
    }

    private fun initVideoFeedView() {
        val videoFeedView = binding.fwVideoFeedView

        val viewOptions = viewOptions {
            baseOptions {
                feedResource(FeedResource.Sku(channelId = FW_CHANNEL_ID, DEFAULT_IDS))
            }
        }

        videoFeedView.init(viewOptions)
    }

    private fun updateViewOptions(skuIds: List<String>) {
        binding.fwVideoFeedView.destroy()
        binding.feedContainer.removeAllViews()
        val videoFeedView = FwVideoFeedView(this)
        val viewOptions = viewOptions {
            baseOptions {
                feedResource(FeedResource.Sku(channelId = FW_CHANNEL_ID, productIds = skuIds))
            }
        }
        binding.feedContainer.addView(videoFeedView)
        videoFeedView.init(viewOptions)
    }

    override fun onDestroy() {
        binding.fwVideoFeedView.destroy()
        super.onDestroy()
    }

    companion object {
        private val DEFAULT_IDS = listOf("6009775390877", "6009775423645", "6009775685789")

        fun intent(context: Context) = Intent(context, SkuActivity::class.java)
    }
}
